specs/models/ArtifactSettings.v1.json

```diff
@@ -0,0 +1,157 @@
+{
+  "title": "ArtifactSettings.v1",
+  "type": "object",
+  "description": "Settings for an artifact",
+  "properties": {
+    "customHook": {
+      "type": "object",
+      "properties": {
+        "module": {
+          "type": "string"
+        },
+        "parameters": {
+          "type": "string"
+        },
+        "user": {
+          "type": "string"
+        },
+        "runBefore": {
+          "type": "boolean"
+        }
+      }
+    },
+    "provenanceGranularityLevel": {
+      "type": "string",
+      "enum": [
+        "off",
+        "coarse",
+        "fine"
+      ]
+    },
+    "permissions": {
+      "type": "string"
+    },
+    "targetFormat": {
+      "type": "string",
+      "enum": [
+        "json",
+        "xml"
+      ]
+    },
+    "targetDatabase": {
+      "type": "string"
+    },
+    "sourceDatabase": {
+      "type": "string"
+    },
+    "additionalCollections": {
+      "type": "array",
```

Comment:
I feel like we don't need "additionalCollections" and can just have "collections". We save default artifact settings with "collections" set when someone creates an artifact. When users update the settings from UI, "collections" gets reset (as collections is not part of the payload or UI property) .

Reply:
We need a way to differentiate between default collections that can't be removed in the UI and user-defined collections that can. Perhaps just have "collections" and "defaultCollections" can be added in on reading the step out?